### PR TITLE
fix(cargo): handle version range using Less Than Equal

### PR DESCRIPTION
**Pull request by caspermeijn**
When a version range with a '<' character is found, the range is updated to the (latest version + 1). However for a range using "<=" this is invalid. Add a special case for a range using Less Than Equal which is updated to simply the latest version.

### CODE REVIEW

File: cargo/spec/dependabot/cargo/update_checker/requirements_updater_spec.rb [hunk 1]
```diff
@@ -151,17 +151,34 @@
...
+            context "when including a pre-release" do
+              let(:req_string) { ">=1.2.0, <1.4.0-dev" }
+              its([:requirement]) { is_expected.to eq(">=1.2.0, <1.6.0") }
+            end
           end
 
-          context "when including a pre-release" do
-            let(:req_string) { ">=1.2.0, <1.4.0-dev" }
-            its([:requirement]) { is_expected.to eq(">=1.2.0, <1.6.0") }
+          context "with `less then equal`" do
```
Comment (bdragon) [inline cargo/spec/dependabot/cargo/update_checker/requirements_updater_spec.rb:169]: Non-blocking/nitpick: technically `less than or equal to`

**Review by bdragon — APPROVED** [review]

File: cargo/lib/dependabot/cargo/update_checker/requirements_updater.rb [hunk 2]
```diff
@@ -141,7 +141,11 @@ def update_range_requirements(string_reqs)
             raise UnfixableRequirement if req.start_with?(">")
 
             req.sub(VERSION_REGEX) do |old_version|
-              update_greatest_version(old_version, target_version)
+              if req.start_with?("<=")
+                version_class.new(target_version)
```
[the remark this answers is not in the document]
Comment (caspermeijn) [reply cargo/lib/dependabot/cargo/update_checker/requirements_updater.rb]: That makes sense. Fixed.

File: cargo/spec/dependabot/cargo/update_checker/requirements_updater_spec.rb [hunk 3]
```diff
@@ -151,17 +151,34 @@
         end
 
         context "when there were multiple range specifications" do
-          let(:req_string) { "> 1.0.0, < 1.2.0" }
-          its([:requirement]) { is_expected.to eq("> 1.0.0, < 1.6.0") }
+          context "with `less then`" do
```
[the remark this answers is not in the document]
Comment (caspermeijn) [reply cargo/spec/dependabot/cargo/update_checker/requirements_updater_spec.rb]: Fixed

File: cargo/spec/dependabot/cargo/update_checker/requirements_updater_spec.rb (hunk 1 above)
Comment (caspermeijn) [reply cargo/spec/dependabot/cargo/update_checker/requirements_updater_spec.rb]: Fixed

**Comment by caspermeijn** [comment]
I fixed all comments and resolved the merge conflict.

**Pull request merged**